maximize-tensorflow-performance-on-cpu-considerations-and-recommendations-for-inference/maximize-tensorflow-performance-on-cpu-considerations-and-recommendations-for-inference.md

```diff
@@ -0,0 +1,287 @@
+</details>
+
+<details>
+  <summary>KMP_AFFINITY</summary>
+  <br>
+  <b>Recommended settings → KMP_AFFINITY=granularity=fine,verbose,compact,1,0</b>
+  <pre>export KMP_AFFINITY=granularity=fine,compact,1,0</pre>
+  tf_cnn_benchmarks usage (shell)
+  <pre>OMP_NUM_THREADS=&lt;number of physical cores per socket&gt; python tf_cnn_benchmarks.py --num_intra_threads=&lt;number of physical cores per socket&gt; --num_inter_threads=&lt;number of sockets&gt; --data_format=NCHW --kmp_affinity=granularity=fine,compact,1,0</pre>
+  
+  Users can bind OpenMP threads to physical processing units. KMP_AFFINITY is used to take advantage of this functionality. It restricts execution of certain threads to a subset of the physical processing units in a multiprocessor computer.
+<br><br>
+Usage of this environment variable is as below.
+<br><br>
+KMP_AFFINITY=[,...][,][,]
+<br><br>
+Modifier is a string consisting of keyword and specifier. type is a string indicating the thread affinity to use. permute is a positive integer value, controls which levels are most significant when sorting the machine topology map. The value forces the mappings to make the specified number of most significant levels of the sort the least significant, and it inverts the order of significance. The root node of the tree is not considered a separate level for the sort operations. offset is a positive integer value, indicates the starting position for thread assignment. We will use the recommended setting of KMP_AFFINITY as an example to explain basic content of this environment variable.
+<br><br>
+KMP_AFFINITY=granularity=fine,verbose,compact,1,0
+<br><br>
+The modifier is granularity=fine,verbose. Fine causes each OpenMP thread to be bound to a single thread context. Verbose prints messages at runtime concerning the supported affinity, and this is optional. These messages include information about the number of packages, number of cores in each package, number of thread contexts for each core, and OpenMP thread bindings to physical thread contexts. Compact is value of type, assigning the OpenMP thread +1 to a free thread context as close as possible to the thread context where the OpenMP thread was placed.
+<br><br>
+NOTE The recommendation changes if Hyperthreading is disabled on your machine. In that case, the recommendation is:   KMP_AFFINITY=granularity=fine,verbose,compact if hyperthreading is disabled.
+<br><br>
+Fig. 2 shows the machine topology map when KMP_AFFINITY is set to these values. The OpenMP thread +1 is bound to a thread context as close as possible to OpenMP thread , but on a different core. Once each core has been assigned one OpenMP thread, the subsequent OpenMP threads are assigned to the available cores in the same order, but they are assigned on different thread contexts.
+<br><br>
+![OpenMP Global Thread Pool IDs](https://www.intel.com/content/dam/develop/external/us/en/images/openmp-global-thread-pool-ids-804042.jpg)
```

Comment:
should replace with AEM image /content/dam/www/central-libraries/us/en/images/openmp-global-thread-pool-ids-804042.jpg

Reply:
Updated